src/common/tags.c

```diff
@@ -398,23 +404,26 @@ static void _tag_execute(const GList *tags, const GList *imgs,
     {
       case DT_TA_ATTACH:
         undotags->after = g_list_copy(undotags->before);
-        _tag_add_tags_to_list(&undotags->after, tags);
+        if(_tag_add_tags_to_list(&undotags->after, tags)) res = TRUE;
         break;
       case DT_TA_DETACH:
         undotags->after = g_list_copy(undotags->before);
-        _tag_remove_tags_from_list(&undotags->after, tags);
+        if(_tag_remove_tags_from_list(&undotags->after, tags)) res = TRUE;
         break;
       case DT_TA_SET:
         undotags->after = g_list_copy((GList *)tags);
         // preserve dt tags
         GList *dttags = _tag_get_tags(image_id, DT_TAG_TYPE_DT);
         if(dttags) undotags->after = g_list_concat(undotags->after, dttags);
+        res = TRUE;
         break;
       case DT_TA_SET_ALL:
         undotags->after = g_list_copy((GList *)tags);
+        res = TRUE;
         break;
       default:
         undotags->after = g_list_copy(undotags->before);
+        res = TRUE;
```

Comment:
after list = before list means there is no change. So no need to signal here.

BTW, comparing the two lists (before / after) is another way to know if there is a change or not. But I don't know if that's more efficient, more effective...

Reply:
of course... done